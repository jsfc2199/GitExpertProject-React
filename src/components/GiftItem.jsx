/* eslint-disable react/prop-types */

import PropTypes from 'prop-types'
const GiftItem = ({title,url}) => {
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GiftItem.propTypes={
  title: PropTypes.string.isRequiered,
  url: PropTypes.string.isRequiered
}
export default GiftItem