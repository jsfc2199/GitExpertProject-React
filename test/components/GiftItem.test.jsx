import { render } from "@testing-library/react";
import GiftItem from "../../src/components/GiftItem";

/* eslint-disable no-undef */
describe('Testing in GiftItem', () => {
  test('should match with snapshot', () => {

    const title= 'This is a title';
    const url = 'https://url.com';

    const {container} = render(<GiftItem title={title} url={url}/>)

    expect(container).toMatchSnapshot()
    
  })
  
})
