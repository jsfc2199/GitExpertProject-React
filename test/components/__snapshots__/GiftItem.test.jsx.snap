// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in GiftItem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="This is a title"
      src="https://url.com"
    />
    <p>
      This is a title
    </p>
  </div>
</div>
`;
